sopel/bot.py

```diff
@@ -132,9 +114,60 @@ def __init__(self, config, daemon=False):
 
     @property
     def command_groups(self):
-        """A mapping of plugin names to lists of their commands."""
+        """A mapping of plugin names to lists of their commands.
+
+        .. versionchanged:: 7.1
+            This attribute is now generated on the fly from the registered list
+            of commands and nickname commands.
+        """
         # This was supposed to be deprecated, but the built-in help plugin needs it
-        return self._command_groups
+        # TODO: create a new, better, doc interface to remove it
```

Comment:
Sure, but I'll have to rebase to do that, because I published this PR without incorporating #1872. So I suggest that we put a pin in that and come back to it later.